### PR TITLE
Upgrade external-snapshotter client to v6.0.1

### PR DESCRIPTION
**Pull request by RaunakShah**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Upgrade external snapshotter client to v6.0.1 release - https://github.com/kubernetes-csi/external-snapshotter/releases/tag/v6.0.1

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Upgrade external-snapshotter client to v6.0.1
```

### CODE REVIEW

**Comment by xing-yang** [comment]
@RaunakShah Please run "go mod vendor".

**Comment by RaunakShah** [comment]
@xing-yang that's weird I thought I had run it already. Pushed in the latest commit.

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/imdario/mergo v0.3.12 // indirect
 	github.com/kubernetes-csi/csi-lib-utils v0.11.0
 	github.com/kubernetes-csi/csi-test/v4 v4.0.2
-	github.com/kubernetes-csi/external-snapshotter/client/v6 v6.0.0-rc4
+	github.com/kubernetes-csi/external-snapshotter/client/v6 v6.0.0
```
Comment (xing-yang) [inline go.mod:13]: Can you change this to 6.0.1 now?

**Pull request merged**